--- addtocal.module	2017-05-02 22:52:07.440453000 -0400
+++ addtocal.module	2017-05-02 22:29:09.168453000 -0400
@@ -598,7 +598,7 @@
   $info = addtocal_extract_event_info($entity_type, $entity, $entity_id, $field_name, $display);

   foreach($info as &$i) {
-    $i = str_replace("\n", "\\n", $i);
+    $i = str_replace("\\n", "", $i); /* Added to remove newline */
   }

   $url = isset($query['url']) ? $query['url'] : '';
@@ -649,6 +649,8 @@
   $display = addtocal_get_display($entity, $entity_type, $field_name, $view_mode);
   $info = addtocal_extract_event_info($entity_type, $entity, $entity_id, $field_name, $display);

+  foreach($info as &$i) {$i = str_replace("\\n", "", $i);} /* Added to remove newline */
+
   $rfc_dates = addtocal_rfc_3339_date($info['start'], $info['end'], $info['timezone']);

   $google_url = url('http://www.google.com/calendar/event', array(
@@ -683,6 +685,8 @@
   $display = addtocal_get_display($entity, $entity_type, $field_name, $view_mode);
   $info = addtocal_extract_event_info($entity_type, $entity, $entity_id, $field_name, $display);

+  foreach($info as &$i) {$i = str_replace("\\n", "", $i);} /* Added to remove newline */
+
   $rfc_dates = addtocal_rfc_3339_date($info['start'], $info['end'], $info['timezone']);

   $yahoo_url = url('http://calendar.yahoo.com/', array(