diff --git a/js/eu_cookie_compliance.js b/js/eu_cookie_compliance.js
index 73a009f..187e7bd 100755
--- a/js/eu_cookie_compliance.js
+++ b/js/eu_cookie_compliance.js
@@ -445,16 +445,18 @@

   // Block cookies when the user hasn't agreed.
   Drupal.behaviors.eu_cookie_compliance_popup_block_cookies = {
+    initialized: false,
     attach: function (context, settings) {
-      $(document).ready(function () {
-        if ((settings.eu_cookie_compliance.method === 'opt_in' && (Drupal.eu_cookie_compliance.getCurrentStatus() === null || !Drupal.eu_cookie_compliance.hasAgreed()))
-          || (settings.eu_cookie_compliance.method === 'opt_out' && !Drupal.eu_cookie_compliance.hasAgreed() && Drupal.eu_cookie_compliance.getCurrentStatus() !== null)
+      if (!Drupal.behaviors.eu_cookie_compliance_popup_block_cookies.initialized && Drupal.settings.eu_cookie_compliance) {
+        Drupal.behaviors.eu_cookie_compliance_popup_block_cookies.initialized = true;
+        if ((Drupal.settings.eu_cookie_compliance.method === 'opt_in' && (Drupal.eu_cookie_compliance.getCurrentStatus() === null || !Drupal.eu_cookie_compliance.hasAgreed()))
+          || (Drupal.settings.eu_cookie_compliance.method === 'opt_out' && !Drupal.eu_cookie_compliance.hasAgreed() && Drupal.eu_cookie_compliance.getCurrentStatus() !== null)
         ) {
           // Split the white-listed cookies.
-          var euCookieComplianceWhitelist = settings.eu_cookie_compliance.whitelisted_cookies.split(/\r\n|\n|\r/g);
+          var euCookieComplianceWhitelist = Drupal.settings.eu_cookie_compliance.whitelisted_cookies.split(/\r\n|\n|\r/g);

           // Add the EU Cookie Compliance cookie.
-          euCookieComplianceWhitelist.push((typeof settings.eu_cookie_compliance.cookie_name === 'undefined' || settings.eu_cookie_compliance.cookie_name === '') ? 'cookie-agreed' : settings.eu_cookie_compliance.cookie_name);
+          euCookieComplianceWhitelist.push((typeof Drupal.settings.eu_cookie_compliance.cookie_name === 'undefined' || Drupal.settings.eu_cookie_compliance.cookie_name === '') ? 'cookie-agreed' : Drupal.settings.eu_cookie_compliance.cookie_name);
           euCookieComplianceBlockCookies = setInterval(function () {
             // Load all cookies from jQuery.
             var cookies = $.cookie();
@@ -496,7 +498,7 @@
             }
           }, 5000);
         }
-      });
+      }
     }
   }
 